ndbench-cockroachdb-plugins/src/main/java/com/netflix/ndbench/plugin/cockroachdb/operations/CockroachDBTransactionPlugin.java

```diff
@@ -0,0 +1,127 @@
+package com.netflix.ndbench.plugin.cockroachdb.operations;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Savepoint;
+import java.sql.Statement;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.netflix.ndbench.api.plugin.annotations.NdBenchClientPlugin;
+import com.netflix.ndbench.plugin.cockroachdb.configs.CockroachDBConfiguration;
+
+/**
+ * @author Sumanth Pasupuleti
+ */
+@Singleton
+@NdBenchClientPlugin("CockroachDBTransactionPlugin")
+public class CockroachDBTransactionPlugin extends CockroachDBPluginBase
+{
+    String readFromMainQuery = "SELECT key, column1, column2, column3, column4 FROM %s where key = ";
+    String writeToMainQuery = "UPSERT INTO %s (key, column1, column2, column3, column4) VALUES ";
+    String writeToChildQuery = "UPSERT INTO %s (key, column1, value) VALUES ";
+
+    @Inject
+    public CockroachDBTransactionPlugin(CockroachDBConfiguration cockroachDBConfiguration) {
+        super(cockroachDBConfiguration);
+    }
+
+    @Override
+    public String readSingle(String key) throws Exception
+    {
+        ResultSet rs = connection.createStatement().executeQuery(readFromMainQuery + "'" + key + "'");
+        int rsSize = 0;
+        while (rs.next())
+        {
+            rsSize++;
+        }
+
+        if (rsSize > 0)
```

Comment:
It seems checking `rsSize>0` is redundant since the statement is not exercised and it will always results in `ResultOK`.

Reply:
it is helpful in detecting CacheMiss

Reply:
A CacheMiss is only exercised in the `rsSize==0` condition.

Reply:
yes, that would be the else block of rsSize>0